src/core/execution/utils/BotBehavior.ts

```diff
@@ -17,6 +17,35 @@ export class BotBehavior {
   private enemy: Player | null = null;
   private enemyUpdated: Tick;
 
+  private shouldBetrayAlly(ally: Player): boolean {
+    const POWER_IMBALANCE_RATIO = 3;
+    const ALLIANCE_OVEREXTENSION = 4;
```

Comment:
Honestly, I think a limit of 4 alliances might be too little. I've seen several multiplayer games where people have up to seven alliances, and theoretically you could get up to 149.
In multiplayer, people generally aren't allying nations, but when I do singleplayer I want to be friendly with as many countries as possible. I'd recommend changing this up to seven:
```suggestion
    const ALLIANCE_OVEREXTENSION = 7;
```

Reply:
I also think it's too little but i tried to reverse the alliance acceptance function, which rejects alliance with players who have more than 4 allies, so theoretically in single player you will never have more than 4 allies.